src/app/components/sideNav/sidenav.services.spec.js

```diff
@@ -0,0 +1,37 @@
+(function() {
+  'use strict';
+  describe('sidenavService', function() {
+    var sidenavService;
+
+    beforeEach( function(){
+      module( 'app.sidenav' );
+
+      inject( function (_sidenavService_) {
+        sidenavService = _sidenavService_;
+      });
+    });
+    it('should contain only functions', function() {
+      expect(sidenavService.toggleList).toEqual(jasmine.any(Function));
+      expect(sidenavService.openLeftMenu).toEqual(jasmine.any(Function));
+      expect(sidenavService.closeSidenav).toEqual(jasmine.any(Function));
+    });
+    it('should not activate any function', function() {
+      spyOn(sidenavService, 'toggleList');
+      expect(sidenavService.toggleList).not.toHaveBeenCalled();
+
+      spyOn(sidenavService, 'openLeftMenu');
+      expect(sidenavService.openLeftMenu).not.toHaveBeenCalled();
+
+      spyOn(sidenavService, 'closeSidenav');
+      expect(sidenavService.closeSidenav).not.toHaveBeenCalled();
+    });
+    it('toggleList should not trigger open/close functions', function() {
+      spyOn(sidenavService, 'openLeftMenu');
+      spyOn(sidenavService, 'closeSidenav');
+
+      sidenavService.toggleList();
+      expect(sidenavService.openLeftMenu).not.toHaveBeenCalled();
+      expect(sidenavService.closeSidenav).not.toHaveBeenCalled();
+    });
```

Comment:
Spend 2 h figuring out why this test doesn't pass.

`
//sidenav.service.js
 function toggleSidenav() {
      console.log($mdSidenav('left').isOpen()); //  console: false
      $mdSidenav(componentId).toggle();
      console.log($mdSidenav('left').isOpen()); //  console: false
      $
    }`


```
//sidenav.service.specs.js
    beforeEach( function(){
      module( 'healthGuide' );
      inject( function (_sidenavService_, _$mdSidenav_) {
        sidenavService = _sidenavService_;
        mdSidenav = _$mdSidenav_;
      });
    });
    it('openLeftMenu should open sideNav', function() {
      expect(mdSidenav.isOpen()).toBeNull();
      sidenavService.toggleList();
      expect($mdSidenav('left').isOpen()).toBeTruthy();
    });|
```